Aztec/Classes/Extensions/NSAttributedString+Lists.swift

```diff
@@ -63,13 +63,82 @@ extension NSAttributedString
     func rangeOfTextList(atIndex index: Int) -> NSRange? {
         var effectiveRange = NSRange()
         let targetRange = rangeOfEntireString
-        guard let _ = attribute(TextList.attributeName, at: index, longestEffectiveRange: &effectiveRange, in: targetRange) as? TextList else {
+        guard
+            let paragraphStyle = attribute(NSParagraphStyleAttributeName, at: index, longestEffectiveRange: &effectiveRange, in: targetRange) as? ParagraphStyle,
+            let _ = paragraphStyle.textList
+        else {
             return nil
         }
 
         return effectiveRange
     }
 
+    /// Returns the range of the given text list that contains the given location.
+    ///
+    /// - Parameter list: The textList to search for.
+    /// - Parameter location: The location in the text list.
+    ///
+    /// - Returns: An NSRange optional containing the range of the list or nil if no list was found.
+    ///
+    func range(of list: TextList,
+               at location: Int) -> NSRange? {
+
+        var effectiveRange = NSRange()
+        let targetRange = rangeOfEntireString
+        guard
+            let paragraphStyle = attribute(NSParagraphStyleAttributeName, at: location, longestEffectiveRange: &effectiveRange, in: targetRange) as? ParagraphStyle,
+            let foundList = paragraphStyle.textList,
+            foundList == list
+        else {
+            return nil
+        }
+        var resultRange = effectiveRange
+        while resultRange.location > 0 {
+            if
+                let paragraphStyle = attribute(NSParagraphStyleAttributeName, at: resultRange.location-1, longestEffectiveRange: &effectiveRange, in: targetRange) as? ParagraphStyle,
+                let foundList = paragraphStyle.textList,
+                foundList == list {
+                resultRange = resultRange.union(withRange: effectiveRange)
+            } else {
+                break;
+            }
+        }
+        while resultRange.endLocation < self.length {
+            if
+                let paragraphStyle = attribute(NSParagraphStyleAttributeName, at: resultRange.endLocation, longestEffectiveRange: &effectiveRange, in: targetRange) as? ParagraphStyle,
+                let foundList = paragraphStyle.textList,
+                foundList == list {
+                resultRange = resultRange.union(withRange: effectiveRange)
+            } else {
+                break;
+            }
+        }
```

Comment:
It returns a correct value, but sometimes the NSParagraphStyle style instance can be different but is the same TextList, so we need to expand the range to grab all the TextList coverage.